python/nistoar/midas/dbio/base.py

```diff
@@ -1018,7 +1025,53 @@ def create_record(self, name: str, shoulder: str = None, foruser: str = None) ->
         rec['name'] = name
         rec = ProjectRecord(self._projcoll, rec, self)
         rec.save()
+        message = f"New {self._projcoll} record created: {name}"
+        self._notify(message)
         return rec
+    
+
+    def _notify(self, message):
+        """
+        Asynchronously sends a notification message via WebSocket.
+        """
+        try:
+            try:
+                loop = asyncio.get_running_loop()
+                logger.info(f"Obtained running event loop: {loop}")
+            except RuntimeError:
+                # No running event loop; create a new one
+                logger.warning("No running event loop found. Creating a new event loop.")
+                loop = asyncio.new_event_loop()
+                asyncio.set_event_loop(loop)
+                logger.info(f"New event loop created: {loop}")
+
+            if not loop.is_running():
+                logger.warning("Event loop is not running. Starting the event loop.")
+                loop.run_until_complete(self._send_notification(message))
+            else:
+                future = asyncio.run_coroutine_threadsafe(self._send_notification(message), loop)
+                logger.info(f"Notification coroutine submitted to the event loop: {future}")
+        except Exception as e:
+            logger.error(f"Unexpected error in _notify: {e}")
+
+    async def _send_notification(self, message):
+        """
+        Coroutine to send a notification message via WebSocket.
+        """
+        uri = "ws://localhost:8765"  # WebSocket server address
```

Comment:
This URL needs to be configurable.  This hard-coded URL assumes a configuration that is set up at the oar-docker level.  If we decide later to run the websocket server in its own container (or on another machine), then we'll need a different URL.